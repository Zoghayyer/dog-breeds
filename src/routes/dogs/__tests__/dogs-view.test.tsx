import React from 'react';
import { shallow } from 'enzyme';
import DogsView, { DogsViewProps } from '../../dogs/dogs-view';

describe('(View) DogsView', () => {
  let component;
  const defaultProps: DogsViewProps = {
    dogsBreedDogs: [],
    favoriteDogs: [],
    isLoading: false,
    showLoadMore: false,
  };

  const shallowDogsView = (extendedProps: Partial<DogsViewProps> = {}) => {
    component = shallow(<DogsView {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowDogsView());

  it('renders dogs view component', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the LoadMore component if showLoadMore is truthy', () => {
    shallowDogsView({ showLoadMore: true });
    expect(component).toMatchSnapshot();
  });

  it('does not render the LoadMore component if showLoadMore is falsy', () => {
    shallowDogsView({ showLoadMore: false });
    expect(component).toMatchSnapshot();
  });

  it('does not render the favorite dogs component if isLoading is true', () => {
    shallowDogsView({ isLoading: true });
    expect(component).toMatchSnapshot();
  });

  it('renders the favorite dogs component if isLoading is false', () => {
    shallowDogsView({ isLoading: false });
    expect(component).toMatchSnapshot();
  });
});
