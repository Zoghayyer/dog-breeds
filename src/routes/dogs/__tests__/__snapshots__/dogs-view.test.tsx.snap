// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(View) DogsView does not render the LoadMore component if showLoadMore is falsy 1`] = `
<div
  className="dogs-container"
>
  <Header
    className="dog-breeds-header"
  >
    <h1>
      Dog Breeds
    </h1>
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
  </Header>
  <Connect(DogsBreedSearchContainer) />
  <Connect(DogsBreedContainer)
    dogs={Array []}
    isLoading={false}
  />
  <Separator
    type="horizontal"
  />
  <Header
    className="favorites-header"
  >
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
    <h1>
      Favorites
    </h1>
  </Header>
  <Connect(DogsBreedContainer)
    dogs={Array []}
  />
</div>
`;

exports[`(View) DogsView does not render the favorite dogs component if isLoading is true 1`] = `
<div
  className="dogs-container"
>
  <Header
    className="dog-breeds-header"
  >
    <h1>
      Dog Breeds
    </h1>
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
  </Header>
  <Connect(DogsBreedSearchContainer) />
  <Connect(DogsBreedContainer)
    dogs={Array []}
    isLoading={true}
  />
  <Separator
    type="horizontal"
  />
  <Header
    className="favorites-header"
  >
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
    <h1>
      Favorites
    </h1>
  </Header>
</div>
`;

exports[`(View) DogsView renders dogs view component 1`] = `
<div
  className="dogs-container"
>
  <Header
    className="dog-breeds-header"
  >
    <h1>
      Dog Breeds
    </h1>
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
  </Header>
  <Connect(DogsBreedSearchContainer) />
  <Connect(DogsBreedContainer)
    dogs={Array []}
    isLoading={false}
  />
  <Separator
    type="horizontal"
  />
  <Header
    className="favorites-header"
  >
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
    <h1>
      Favorites
    </h1>
  </Header>
  <Connect(DogsBreedContainer)
    dogs={Array []}
  />
</div>
`;

exports[`(View) DogsView renders the LoadMore component if showLoadMore is truthy 1`] = `
<div
  className="dogs-container"
>
  <Header
    className="dog-breeds-header"
  >
    <h1>
      Dog Breeds
    </h1>
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
  </Header>
  <Connect(DogsBreedSearchContainer) />
  <Connect(DogsBreedContainer)
    dogs={Array []}
    isLoading={false}
  />
  <Connect(LoadMoreContainer) />
  <Separator
    type="horizontal"
  />
  <Header
    className="favorites-header"
  >
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
    <h1>
      Favorites
    </h1>
  </Header>
  <Connect(DogsBreedContainer)
    dogs={Array []}
  />
</div>
`;

exports[`(View) DogsView renders the favorite dogs component if isLoading is false 1`] = `
<div
  className="dogs-container"
>
  <Header
    className="dog-breeds-header"
  >
    <h1>
      Dog Breeds
    </h1>
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
  </Header>
  <Connect(DogsBreedSearchContainer) />
  <Connect(DogsBreedContainer)
    dogs={Array []}
    isLoading={false}
  />
  <Separator
    type="horizontal"
  />
  <Header
    className="favorites-header"
  >
    <Icon
      alt="red heart icon"
      icon="redHeartIcon"
    />
    <h1>
      Favorites
    </h1>
  </Header>
  <Connect(DogsBreedContainer)
    dogs={Array []}
  />
</div>
`;
